$NetBSD$

Keep DH group strings and KEX names read-only, as in NetBSD.

--- dh.c.orig
+++ dh.c
@@ -352,7 +352,7 @@
 DH *
 dh_new_group1(void)
 {
-	static char *gen = "2", *group1 =
+	static const char *gen = "2", *group1 =
 	    "FFFFFFFF" "FFFFFFFF" "C90FDAA2" "2168C234" "C4C6628B" "80DC1CD1"
 	    "29024E08" "8A67CC74" "020BBEA6" "3B139B22" "514A0879" "8E3404DD"
 	    "EF9519B3" "CD3A431B" "302B0A6D" "F25F1437" "4FE1356D" "6D51C245"
@@ -367,7 +367,7 @@
 DH *
 dh_new_group14(void)
 {
-	static char *gen = "2", *group14 =
+	static const char *gen = "2", *group14 =
 	    "FFFFFFFF" "FFFFFFFF" "C90FDAA2" "2168C234" "C4C6628B" "80DC1CD1"
 	    "29024E08" "8A67CC74" "020BBEA6" "3B139B22" "514A0879" "8E3404DD"
 	    "EF9519B3" "CD3A431B" "302B0A6D" "F25F1437" "4FE1356D" "6D51C245"
@@ -387,7 +387,7 @@
 DH *
 dh_new_group16(void)
 {
-	static char *gen = "2", *group16 =
+	static const char *gen = "2", *group16 =
 	    "FFFFFFFF" "FFFFFFFF" "C90FDAA2" "2168C234" "C4C6628B" "80DC1CD1"
 	    "29024E08" "8A67CC74" "020BBEA6" "3B139B22" "514A0879" "8E3404DD"
 	    "EF9519B3" "CD3A431B" "302B0A6D" "F25F1437" "4FE1356D" "6D51C245"
@@ -418,7 +418,7 @@
 DH *
 dh_new_group18(void)
 {
-	static char *gen = "2", *group18 =
+	static const char *gen = "2", *group18 =
 	    "FFFFFFFF" "FFFFFFFF" "C90FDAA2" "2168C234" "C4C6628B" "80DC1CD1"
 	    "29024E08" "8A67CC74" "020BBEA6" "3B139B22" "514A0879" "8E3404DD"
 	    "EF9519B3" "CD3A431B" "302B0A6D" "F25F1437" "4FE1356D" "6D51C245"
--- kex-names.c.orig
+++ kex-names.c
@@ -44,7 +44,7 @@
 #include "ssherr.h"
 
 struct kexalg {
-	char *name;
+	const char *name;
 	u_int type;
 	int ec_nid;
 	int hash_alg;
